File: gpao-tex-ui/source/main/src/app/services/order/models/paquet.ts
/* tslint:disable */
/* eslint-disable */
import { FicheEclatement } from '../models/fiche-eclatement';
import { OrdreFabrication } from '../models/ordre-fabrication';
export interface Paquet {
  archived?: boolean;
  couleurId?: number;
  creationDate?: string;
  creatorIdentifier?: string;
  deletionDate?: string;
  engagement?: boolean;
  entreeAssemblage?: boolean;
  entreeDevant?: boolean;
  entreeDos?: boolean;
  entreeFinition?: boolean;
  entreePreparation?: boolean;
  ficheEclatement?: FicheEclatement;
  id?: number;
  latestUpdaterId?: string;
  modificationDate?: string;
  num?: number;
  numMatelas?: string;
  numRouleaux?: string;
  ordreFabrication?: OrdreFabrication;
  productionHeure?: boolean;
  quantite?: number;
  retrait?: string;
  siteId?: number;
  sortieAssemblage?: boolean;
  sortieDevant?: boolean;
  sortieDos?: boolean;
  sortiePreparation?: boolean;
  tailleId?: number;
  version?: number;
}

File: gpao-tex-core/ordre/src/main/java/com/gpro/consulting/ordre/model/Paquet.java
package com.gpro.consulting.ordre.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "gp_paquet")
@Data
@EqualsAndHashCode(callSuper = true)
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Paquet extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "gp_of_id")
    private OrdreFabrication ordreFabrication;

    @ManyToOne
    @JoinColumn(name = "ficheEclatement_id")
    private FicheEclatement ficheEclatement;

    @Id
    @GeneratedValue(generator = "FICHE_ECLATEMENT_ID_GENERATOR", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "FICHE_ECLATEMENT_SEQ", sequenceName = "FICHE_ECLATEMENT_SEQ", allocationSize = 1)
    private Long id;
    private Long num;
    private Boolean productionHeure;
    private Long siteId;
    @Column(name = "eb_couleur_id")
    private Long couleurId;
    @Column(name = "eb_taille_id")
    private Long tailleId;
    private Long quantite;
    private String numMatelas;
    private String numRouleaux;
    private Boolean entreeDos;
    private Boolean sortieDos;
    private Boolean entreeDevant;
    private Boolean sortieDevant;
    private Boolean sortieAssemblage;
    private Boolean entreeAssemblage;
    private Boolean entreeFinition;
    private Boolean entreePreparation;
    private Boolean sortiePreparation;
    private Boolean engagement;
    private String retrait;
}
